include/exadg/incompressible_navier_stokes/spatial_discretization/operator_dual_splitting.cpp

```diff
@@ -801,7 +801,10 @@ OperatorDualSplitting<dim, Number>::local_interpolate_velocity_dirichlet_bc_boun
   VectorType const &,
   Range const & face_range) const
 {
-  unsigned int const dof_index  = this->get_dof_index_velocity();
+  unsigned int const dof_index = this->get_dof_index_velocity();
+  AssertThrow(
+    matrix_free.get_dof_handler(dof_index).get_triangulation().all_reference_cells_are_hyper_cube(),
+    dealii::ExcMessage("This function is not supported for simplices."));
```

Comment:
Thanks for adding asserts here.

It would be slightly more precise to say that the function is only implemented for pure hypercube elements currently. There might be more variants in the future and, then, the current error message might be unclear.